widgetssdk/src/main/java/com/glia/widgets/di/SharedDependencyFactory.kt

```diff
@@ -0,0 +1,19 @@
+package com.glia.widgets.di
+
+import com.glia.widgets.chat.ChatManager
+
+internal class SharedDependencyFactory(private val useCaseFactory: UseCaseFactory) {
```

Comment:
Could you please explain the purpose of `SharedDependencyFactory`? Do I understand correctly that it's needed for cases when some dependency should be initialized after the others which it depends on?

Reply:
Since we have factories for UseCases, Repositories, Controllers an so on, so I created factory for the classes which are not among those listed above.

Reply:
Ah, got it now. "Shared" confused me.

Reply:
If have a better name suggestion, I'd like to rename it to something more descriptive.

Reply:
Maybe `ManagerFactory`?

Reply:
Why not. I was thinking of creating something common for all other classes, But at the moment we have only this ChatManager class.

Reply:
fixed - fa506584650c3ca6870b22f4570256ea193ea018